tests/testsuite/build_script_env.rs

```diff
@@ -5,6 +5,109 @@ use cargo_test_support::project;
 use cargo_test_support::sleep_ms;
 use cargo_test_support::str;
 
+#[cargo_test]
+fn rerun_if_env_changes_config() {
```

Comment:
> btw I believe this would also fix https://github.com/rust-lang/cargo/issues/8693.

Since we should be getting that fix "for free", could you add a test for a cargo-set env variable (don't think it has to literally be that one) and mark this PR as fixing that issue as well?

Reply:
It has been added and you can continue to review it.